lib/ansible/modules/cloud/azure/azure_rm_sql_server.py

```diff
@@ -0,0 +1,292 @@
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif key == "location":
+                self.parameters["location"] = kwargs[key]
+            elif key == "tags":
+                self.parameters["tags"] = kwargs[key]
+            elif key == "identity":
+                self.parameters["identity"] = kwargs[key]
+            elif key == "administrator_login":
+                self.parameters["administrator_login"] = kwargs[key]
+            elif key == "administrator_login_password":
+                self.parameters["administrator_login_password"] = kwargs[key]
+            elif key == "version":
+                self.parameters["version"] = kwargs[key]
+
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {0} not found'.format(self.resource_group))
+
+        response = self.get_servers()
+
+        if not response:
+            self.log("Server instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Server instance already exists")
+            if self.state == 'absent':
+                self.delete_servers()
+                self.results['changed'] = True
+                self.log("Server instance deleted")
+            elif self.state == 'present':
+                self.log("Need to check if Server instance has to be deleted or may be updated")
+                to_be_updated = True
+
+        if self.state == 'present':
+
+            self.log("Need to Create / Update the Server instance")
+
+            if self.check_mode:
+                return self.results
+
+            if to_be_updated:
+                self.results['state'] = self.create_update_servers()
+                self.results['changed'] = True
+            else:
+                self.results['state'] = response
```

Comment:
We should look at changing the return values, see my note in the documentation portion.